lang/lowering/src/lib.rs

```diff
@@ -6,21 +6,29 @@ mod result;
 use ast::{self};
 use parser::cst;
 
-use crate::lookup_table::build_lookup_table;
 use crate::lower::Lower;
 
 pub use ctx::*;
+pub use lookup_table::{build_lookup_table, LookupTable};
 pub use result::*;
 
 pub fn lower_module(prg: &cst::decls::Module) -> Result<ast::Module, LoweringError> {
-    let cst::decls::Module { uri, use_decls, decls } = prg;
+    lower_module_with_lookup_table(prg, &mut LookupTable::default())
+}
+
+pub fn lower_module_with_lookup_table(
+    prg: &cst::decls::Module,
+    lookup_table: &mut LookupTable,
+) -> Result<ast::Module, LoweringError> {
+    let mut combined_table = std::mem::take(lookup_table);
+    combined_table.append(build_lookup_table(prg)?);
 
-    let lookup_table = build_lookup_table(prg)?;
+    let mut ctx = Ctx::empty(combined_table);
 
-    let mut ctx = Ctx::empty(lookup_table);
+    let use_decls = prg.use_decls.lower(&mut ctx)?;
+    let decls = prg.decls.lower(&mut ctx)?;
 
-    let use_decls = use_decls.lower(&mut ctx)?;
-    let decls = decls.lower(&mut ctx)?;
+    *lookup_table = ctx.lookup_table;
```

Comment:
What does this line do?

Reply:
I guess it is somehow related to the `std::mem::take`? It is not obvious what the API is w.r.t. to the function: Will the function return the lookup_table unchanged, or add the elements contained in the module?

Reply:
`Ctx` expects a `LookupTable` which it then proceeds to own. Hence, we convert the `&mut LookupTable` to an owned one using `std::mem::take`. After lowering, we move the `lookup_table` out of the context and assign the modified version back to the mutable variable.